pepper-apis/src/main/java/org/broadinstitute/ddp/util/StatisticsUtil.java

```diff
@@ -0,0 +1,94 @@
+import org.broadinstitute.ddp.db.dao.JdbiQuestionCached;
+import org.broadinstitute.ddp.db.dao.QuestionDao;
+import org.broadinstitute.ddp.db.dao.StatisticsConfigurationDao;
+import org.broadinstitute.ddp.db.dto.PicklistOptionDto;
+import org.broadinstitute.ddp.db.dto.QuestionDto;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.json.statistics.StatisticsResponse;
+import org.broadinstitute.ddp.json.statistics.StatisticsResponseItem;
+import org.broadinstitute.ddp.model.activity.instance.question.PicklistOption;
+import org.broadinstitute.ddp.model.statistics.StatisticsConfiguration;
+import org.broadinstitute.ddp.model.statistics.StatisticsTypes;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.stringtemplate4.StringTemplateSqlLocator;
+
+public class StatisticsUtil {
+    //private static final Logger LOG = LoggerFactory.getLogger(StatisticsUtil.class);
+    private static final String QUERIES_PATH = "org/broadinstitute/ddp/db/dao/Statistics.sql.stg";
+
+    private static final Map<StatisticsTypes, String> STATISTICS_QUERIES = new HashMap<>();
+
+    static {
+        STATISTICS_QUERIES.put(StatisticsTypes.PARTICIPANTS, "getParticipantStatistics");
+        STATISTICS_QUERIES.put(StatisticsTypes.DISTRIBUTION, "getAnswersDistributionByQueryStableId");
+        STATISTICS_QUERIES.put(StatisticsTypes.MAILING_LIST, "getMailingListStatistics");
+        STATISTICS_QUERIES.put(StatisticsTypes.SPECIFIC_ANSWER, "getSpecificAnswerStatistics");
+        STATISTICS_QUERIES.put(StatisticsTypes.KITS, "getKitStatistics");
+    }
+
+    public static List<StatisticsResponse> generateStatisticsForStudy(Handle handle, StudyDto study,
+                                                                      I18nContentRenderer renderer,
+                                                                      ContentStyle style, long langId) {
+        StatisticsConfigurationDao statConfigDao = handle.attach(StatisticsConfigurationDao.class);
+        List<StatisticsConfiguration> statConfigs = statConfigDao.getStatisticsConfigurationForStudy(study.getId());
+        List<StatisticsResponse> studyStatistics = new ArrayList<>();
+        for (StatisticsConfiguration statConfig : statConfigs) {
+            studyStatistics.add(generateStatisticsItem(handle, study.getUmbrellaId(), renderer, statConfig, style, langId));
+        }
+        return studyStatistics;
+    }
+
+    private static StatisticsResponse generateStatisticsItem(Handle handle, long umbrellaId,
+                                                             I18nContentRenderer renderer,
+                                                             StatisticsConfiguration statConfig,
+                                                             ContentStyle style, long langId) {
+        List<StatisticsResponseItem> statisticsItems;
+        String query = StringTemplateSqlLocator.findStringTemplate(QUERIES_PATH, STATISTICS_QUERIES.get(statConfig.getType())).render();
```

Comment:
The typical usage pattern is either:
1) Name your `.stg` file with same name as this class and in the same folder structure as this class. Then use `.findStringTemplate(YourClass.class, "queryName")`.
2) Or just put it in a DAO interface that extends `SqlObject`.

Reply:
I know about different approaches and this one is working as well so your comment is not clear for me: should I rewrite the code?

Reply:
Yes, please change the code -- I recommend option 1. It's preferable to follow patterns used in the codebase.

Reply:
Changed to option #1